### ISSUE TITLE
Why is ref_per_token_logps - per_token_logps clipped to -10 and 10?

### ISSUE THREAD

**Issue by Parsifal133** [tulerfeng/Video-R1#67]
Thank you for your awesome work！

I noticed that in your code, the token_logps of the two models are clipped to -10 and 10, that is：

```
x_clamped = torch.clamp(ref_per_token_logps - per_token_logps min=-10, max=10) 
per_token_kl = torch.exp(x_clamped) - x_clamped - 1.
```

As far as I know, the value of x_clamped has a certain important impact on training. Setting this value too large or too small may lead to suboptimal training results. Do you have any suggestions for tuning parameters?

Actually this question comes from my recent experiment. I used your code and data to train a video LLM similar to LLaVA, but it is difficult to obtain a smoothly rising reward curve (**the orange curve in the figure below**). I reproduced your code, where the KL divergence rises gently. 
In fact, the following model, which was trained for 3200 iterations, achieved a 7% improvement on the MVSD open question-answering dataset compared to the cold start model, but did not bring consistent improvement on the MvBench dataset.

<img width="351" height="247" alt="Image" src="https://github.com/user-attachments/assets/cfb05da6-9283-4c3e-9614-e22dbaa9227e" />
<img width="350" height="240" alt="Image" src="https://github.com/user-attachments/assets/151f41eb-a1d8-449c-becf-2bd3e8a19747" />

When I set the clipping value to **-1 and 1**, the kl curve will rise smoothly, but after several iterations of training, the reward of the model will drop off a cliff (**the gray curve is in the figure below**).

<img width="356" height="256" alt="Image" src="https://github.com/user-attachments/assets/ac35fbe6-5bda-4d63-b9c5-5b5dcaafc1ed" />
<img width="349" height="250" alt="Image" src="https://github.com/user-attachments/assets/0e2d71c8-65b6-4fb6-9b31-5fe948819be8" />

I would really appreciate any suggestions

**Comment by Parsifal133** [comment]
Here is my reproduction of the KL divergence and reward curve, using your code, model and data
<img width="356" height="260" alt="Image" src="https://github.com/user-attachments/assets/e381c30a-04b4-4b70-858d-2d9fe7ca67da" />
<img width="350" height="255" alt="Image" src="https://github.com/user-attachments/assets/40d63260-5082-44f5-ac3a-d2683dae0819" />

**Comment by tulerfeng** [comment]
Hi, thanks for your question.

Based on my experience, setting the clipping value too large can lead to numerical instability due to the exponential operation, potentially causing training to explode. On the other hand, setting it too small can result in an inaccurate approximation of the KL divergence, reducing training stability in reinforcement learning. When the value is set to 1, the constraint may be too weak, leading to uncontrolled exploration and possible training collapse. In my case, setting it to 10 has generally worked well.

You might also find this blog post helpful, which discusses KL divergence in GRPO training (in Chinese):
https://zhuanlan.zhihu.com/p/25862547100

A few additional suggestions:

1. Besides tuning this clipping parameter, you might also adjust the beta value, which controls the weight of the KL term.

2. Limited improvements on MvBench can result from various factors: training data quality, base model selection, training recipes, or the number of frames used during training and evaluation. Consider filtering or adding higher-quality data for RL training or SFT. Also, note that some MvBench questions may not require reasoning, so incorporating an adaptive reasoning strategy might help.

3. For better visualization in Wandb, you can use the smoothing button on training curves (adjusting the y-axis range can also help). Additionally, instead of tracking the overall reward, monitoring the Accuracy reward might provide more meaningful insights.
